### PR TITLE
first version of ownersvoters logic

### PR DESCRIPTION
**Pull request by MaximeZmt**
Thank you for opening a pull request with this project, please also:

* [ ] add a brief description of your changes here
* [ ] assign the PR to yourself, or to the person(s) working on it
* [ ] start in `draft` mode and `in progress` pipeline in the project (if applicable)
* [ ] if applicable, add this PR to its related issue by one of the special keywords [Closing keywords](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue)
* once it's ready 
  * [ ] put it in the `Review` or `Ready4Merge` pipeline in the project (if applicable) 
  * [ ] remove `draft`
  * [ ] assign a reviewer

### CODE REVIEW

File: contracts/evoting/evoting.go [hunk 1]
```diff
@@ -854,6 +927,30 @@ func (e evotingCommand) manageAdminList(snap store.Snapshot, step execution.Step
...
+	if err != nil {
+		return false, xerrors.Errorf("Failed to convert SCIPER to int: %v", err)
+	}
+
+	if role == Voters {
+		for i := 0; i < len(form.Voters); i++ {
+			if form.Voters[i] == sciperInt {
+				return true, nil
+			}
+		}
```
Comment (PascalinDe) [inline contracts/evoting/evoting.go:942]: the list of voters can be quite long (e.g. 13 000 people at EPFL) - I wonder if it'd make sense to maintain the voters as a sorted list

that would make inserting new voters a bit more expensive, but looking up existing voters much less expensive

what do you think?

File: contracts/evoting/mod_test.go [hunk 2]
```diff
@@ -1253,23 +1263,14 @@ func TestCommand_AdminList(t *testing.T) {
...
-	addOwner := types.AddOwner{
-		FormID:           fakeFormID,
-		TargetUserID:     dummyUserAdminID,
-		PerformingUserID: dummyUserAdminID,
-	}
-
-	// Test Serialization of AddOwner command
-	dataAdd, err := addOwner.Serialize(ctx)
-	require.NoError(t, err)
+	dummyUser := 654321
```
Comment (PascalinDe) [inline contracts/evoting/mod_test.go:1273]: this does not seem to be used

File: integration/ballot.go [hunk 3]
```diff
@@ -57,6 +57,32 @@ func castVotesRandomly(m txManager, actor dkg.Actor, form types.Form,
 
 	votes := make([]types.Ballot, numberOfVotes)
 
+	for i := 0; i < numberOfVotes; i++ {
+		voterID := strconv.Itoa(i+1) + "11111"
+		voterID = voterID[:6]
```
Comment (PascalinDe) [inline integration/ballot.go:62]: it's a bit unclear to me what this does

File: integration/integration_test.go [hunk 4]
```diff
@@ -391,14 +391,14 @@ func getIntegrationBenchmark(numNodes, numVotes int) func(*testing.B) {
 		err = waitForStatus(types.Closed, formFac, formID, nodes, numNodes,
 			5*time.Second)
 		require.NoError(b, err)
-
+		time.Sleep(time.Second * 5)
```
Comment (PascalinDe) [inline integration/integration_test.go]: why do we need to wait here?

Comment (MaximeZmt) [reply integration/integration_test.go]: Oh I need to remove it, I forgot to do so. will do👍

File: integration/ballot.go (hunk 3 above)
Comment (MaximeZmt) [reply integration/ballot.go:62]: Basically, it depending the test calling this method, it is going to generate a massive number of votes. I need to create a voter id for each of these voters. So I decided to use the ith voter as an id.
But as they are store as number and range from 100000 to 999999 (representing sciper)
and as I don't know how many voters they will be created in the method,
I pad the ith number with 1's and then I truncate to take the first six number.

if it generate 10 voters, i is going to vary from 0 to 9
with the +1
from 1 to 10
we are going to get id:
1 11111
2 11111
3 11111
4 11111
5 11111
6 11111
7 11111
8 11111
9 11111
10 11111 -> truncate to 6 digits: 101111

is my expaination clear enough ? 😅

Comment (PascalinDe) [reply integration/ballot.go:62]: OK, could you add a brief comment in the code explaining this?

Comment (MaximeZmt) [reply integration/ballot.go:62]: will do 👍

File: contracts/evoting/evoting.go (hunk 1 above)
Comment (MaximeZmt) [reply contracts/evoting/evoting.go:942]: image
after having performed some benchmarking trying to assess the cost of accessing values in a Go Slice, I got the result above.
Therefore assuming 1ns per access to the slice, it makes ~0.000013sec to go accross the 13 000 people at EPFL which in my opinion does not seems too problematic.

**Pull request merged**